src/data/reports/glass-admin/GLASSDataMaintenanceDefaultRepository.ts

```diff
@@ -0,0 +1,195 @@
+import _ from "lodash";
+import { Config } from "../../../domain/common/entities/Config";
+import { Id } from "../../../domain/common/entities/Base";
+import { Paging, Sorting } from "../../../domain/common/entities/PaginatedObjects";
+
+export class GLASSDataMaintenanceDefaultRepository implements GLASSDataMaintenanceRepository {
+    private storageClient: StorageClient;
+    private globalStorageClient: StorageClient;
+
+    constructor(private api: D2Api) {
+        const instance = new Instance({ url: this.api.baseUrl });
+        this.storageClient = new DataStoreStorageClient("user", instance);
+        this.globalStorageClient = new DataStoreStorageClient("global", instance);
+    }
+
+    async get(
+        options: GLASSDataMaintenanceOptions,
+        namespace: string
+    ): Promise<GLASSMaintenancePaginatedObjects<GLASSDataMaintenanceItem>> {
+        const { paging, sorting, module } = options;
+        if (!module) return emptyPage;
+
+        const uploads = await this.getUploads(namespace);
+        const orgUnits = await this.getCountries();
+
+        const rows = this.getFilteredFiles(uploads, orgUnits, module);
+
+        const rowIds = this.getRowIds(rows);
+        const { objects, pager } = this.paginate(rows, sorting, paging);
+
+        return { objects: objects, pager: pager, rowIds: rowIds };
+    }
+
+    async getUserModules(config: Config): Promise<GLASSModule[]> {
+        const modules = await this.getModules();
+
+        const userGroups = config.currentUser.userGroups;
+        const userGroupIds = userGroups.map(userGroup => userGroup.id);
+
+        const userModules = modules.filter(module => {
+            const moduleUserGroupIds = module.userGroups.approveAccess?.map(userGroup => userGroup.id) ?? [];
+
+            return _.some(moduleUserGroupIds, moduleUserGroupId => userGroupIds.includes(moduleUserGroupId));
+        });
+
+        return userModules;
+    }
+
+    async delete(namespace: string, items: Id[]): Promise<void> {
+        await this.deleteFileResources(items);
+
+        const uploads = await this.getUploads(namespace);
+        const updatedUploads = this.updateStatusById(items, uploads, "DELETED");
+        return await this.globalStorageClient.saveObject<GLASSDataMaintenanceItem[]>(namespace, updatedUploads);
+    }
+
+    async getColumns(namespace: string): Promise<string[]> {
+        const columns = await this.storageClient.getObject<string[]>(namespace);
+
+        return columns ?? [];
+    }
+
+    async saveColumns(namespace: string, columns: string[]): Promise<void> {
+        return this.storageClient.saveObject<string[]>(namespace, columns);
+    }
+
+    private async getCountries(): Promise<NamedRef[]> {
+        const { organisationUnits } = await this.api
+            .get<{ organisationUnits: NamedRef[] }>("/organisationUnits", {
+                fields: "id, name",
+                filter: "level:eq:3",
+                paging: false,
+            })
```

Comment:
We should use `this.api.models.organisationUnits` or `this.api.metadata.get({organisationUnits: ...})`